koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -881,65 +860,236 @@ def handler(self, src, target, opts=None):
+        )
+
+        results = []
+        semi_results = self.createContainer(**kwargs)
+        if semi_results is not None:
+            results = [semi_results]
+
+        self.logger.debug("Results: %r", results)
+
+        all_repositories = []
+        all_koji_builds = []
+
+        if not results:
+            return {
+                'repositories': all_repositories,
+                'koji_builds': all_koji_builds,
+                'build': 'skipped',
+            }
+
+        for result in results:
+            try:
+                repository = result.get('repositories')
+                all_repositories.extend(repository)
+            except Exception as error:
+                self.logger.error("Failed to merge list of repositories "
+                                  "%r. Reason (%s): %s", repository,
+                                  type(error), error)
+            koji_build_id = result.get('koji_build_id')
+            if koji_build_id:
+                all_koji_builds.append(koji_build_id)
+
+        return {
+            'repositories': all_repositories,
+            'koji_builds': all_koji_builds,
+        }
+
+
+class BuildSourceContainerTask(BaseContainerTask):
+    """Start builds via osbs"""
+    Methods = ['buildSourceContainer']
+    # Same value as for regular 'build' method.
+    _taskWeight = 2.0
+
+    # JSON Schema definition for koji buildSourceContainer task parameters
+    # Used to validate arguments passed to the handler() method of this class
+    PARAMS_SCHEMA = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "description": "Parameters for a koji buildSourceContainer task.",
+
+        "type": "array",
+        "items": [
+            {
+                "type": "string",
+                "description": "Build target."
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources",
+                        "examples": [1233, 1234, 1235]
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources",
+                        "examples": [
+                           "some_image_build-3.0-30",
+                           "another_image_build-4.0-10"
+                        ]
+                    },
+                    "signing_intent": {
+                        "type": ["string"],
+                        "description": "Signing intent of the ODCS composes."
+                    },
+                },
+                "anyOf": [
+                    {"required": ["koji_build_id", "koji_build_nvr"]},
```

Comment:
I don't think we need this line. If both are provided, both the other 'required' schema will also validate.

Reply:
right